interface.go

```diff
@@ -2,44 +2,63 @@ package sectorbuilder
 
 import (
 	"context"
+	"errors"
+	"fmt"
 	"io"
 
-	ffi "github.com/filecoin-project/filecoin-ffi"
 	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-storage/storage"
 	"github.com/ipfs/go-cid"
+)
+
+type SectorFileType int
 
-	"github.com/filecoin-project/go-sectorbuilder/fs"
+const (
+	FTUnsealed SectorFileType = 1 << iota
+	FTSealed
+	FTCache
 )
 
-type Interface interface {
-	RateLimit() func()
+func (t SectorFileType) String() string {
+	switch t {
+	case FTUnsealed:
+		return "unsealed"
+	case FTSealed:
+		return "sealed"
+	case FTCache:
+		return "cache"
+	default:
+		return fmt.Sprintf("<unknown %d>", t)
+	}
+}
 
-	AddPiece(context.Context, abi.UnpaddedPieceSize, abi.SectorNumber, io.Reader, []abi.UnpaddedPieceSize) (abi.PieceInfo, error)
-	SectorSize() abi.SectorSize
-	SealProofType() abi.RegisteredProof
-	PoStProofType() abi.RegisteredProof
-	AcquireSectorNumber() (abi.SectorNumber, error)
-	Scrub([]abi.SectorNumber) []*Fault
+type SectorPaths struct {
+	Id abi.SectorID
```

Comment:
Opinion: I think you should use `abi.SectorNumber` here instead of `abi.SectorID`.

Reply:
I want to transition most functions here to SectorID to make things like multi-miner workers simpler to implement (and just much cleaner code-wise)